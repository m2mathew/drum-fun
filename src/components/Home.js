// External Dependencies
import React from 'react';
import { Text, StyleSheet, View } from 'react-native';

// Internal Dependencies
import theme from './theme';

// Component Definition
const Home = () => (
  <View style={styles.home}>
    <Text style={styles.text}>DrumFun</Text>
  </View>
);

const styles = StyleSheet.create({
  home: {
    flex: 1,
  },
  text: {
    fontFamily: 'Changa-Regular',
    fontSize: 40,
    fontWeight: '500',
    textAlign: 'center',
    color: theme.black,
  },
});

export default Home;
