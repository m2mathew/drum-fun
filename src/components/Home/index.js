// External Dependencies
import React from 'react';
import { Button, Text, StyleSheet, View } from 'react-native';

// Internal Dependencies
import theme from '../theme';

// Component Definition
const Home = () => {
  function handleClickButton() {
    console.log('button pressed!');
  }

  return (
    <View style={styles.home} testID="home-wrapper">
      <Text style={styles.text}>DrumFun</Text>
      <Button
        title="Load a Rudiment"
        onPress={handleClickButton}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  home: {
    flex: 1,
  },
  text: {
    fontFamily: 'Changa-Regular',
    fontSize: 40,
    fontWeight: '500',
    textAlign: 'center',
    color: theme.black,
  },
});

export default Home;
