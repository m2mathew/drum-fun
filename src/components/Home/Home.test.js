// External Dependencies
import React from 'react';
import { fireEvent, render, wait } from '@testing-library/react-native';

// Internal Dependencies
import Home from './index';

// Begin Tests
test('should render Home', async () => {
  // const { getByText } = render(<Home {...props} />);
  const { baseElement, getByText, queryByTestId } = render(<Home />);

  console.log('baseElement', baseElement);

  const button = getByText('Load a Rudiment');
  fireEvent.press(button);

  await wait(() => expect(queryByTestId('printed-username')).toBeTruthy());

  expect(baseElement).toMatchSnapshot();

  expect(getByTestId('home-wrapper').props.children).toHaveLength(1);
});
