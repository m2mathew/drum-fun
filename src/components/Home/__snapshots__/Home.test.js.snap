// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
      testID="home-wrapper"
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontFamily": "Changa-Regular",
            "fontSize": 40,
            "fontWeight": "500",
            "textAlign": "center",
          }
        }
      >
        DrumFun
      </Text>
    </View>
  </View>
</View>
`;
